src/core/Akka/Actor/ActorRefProvider.cs

```diff
@@ -341,67 +343,96 @@ private ActorRef ResolveActorRef(InternalActorRef actorRef, IReadOnlyCollection<
+                        props2 = props2.WithDispatcher(propsDeploy.Dispatcher);
+                }
+
+                //TODO: implement Dispatchers and uncomment
+                //if (!system.Dispatchers.HasDispatcher(props2.Dispatcher))
+                //{
+                //    throw new ConfigurationException(string.Format("Dispatcher [{0}] not configured for path {1}", props2.Dispatcher, path));
+                //}
+
+                try
+                {
+                    // for consistency we check configuration of dispatcher and mailbox locally
+                    var dispatcher = _system.Dispatchers.Lookup(props2.Dispatcher);
+                    var mailboxType = _system.Mailboxes.GetMailboxType(props2, ConfigurationFactory.Empty);
+                    var mailbox = _system.Mailboxes.CreateMailbox(props2, ConfigurationFactory.Empty);
+                    //TODO: dispatcher need configurators
+
+                    if (async)
+                        return
+                            new RepointableActorRef(system, props2, dispatcher,
+                                () => _system.Mailboxes.CreateMailbox(props2, ConfigurationFactory.Empty), supervisor,
+                                path).Initialize(async);
+                    return new LocalActorRef(system, props2, dispatcher,
+                        () => _system.Mailboxes.CreateMailbox(props2, ConfigurationFactory.Empty), supervisor, path);
+                }
+                catch (Exception ex)
+                {
+                    throw new ConfigurationException(
+                        string.Format(
+                            "Configuration problem while creating {0} with dispatcher [{1}] and mailbox [{2}]", path,
+                            props.Dispatcher, props.Mailbox), ex);
+                }
             }
-
-            return CreateWithRouter(system, props, supervisor, path, deploy, async);
-        }
-
-        private InternalActorRef CreateWithRouter(ActorSystemImpl system, Props props, InternalActorRef supervisor,
-            ActorPath path, Deploy deploy, bool async)
-        {
-            //if no Router config value was specified, override with procedural input
-            if (deploy.RouterConfig.NoRouter())
+            else //routers!!!
             {
-                deploy = deploy.WithRouterConfig(props.RouterConfig);
-            }
-
-            var routerDispatcher = system.Dispatchers.FromConfig(props.RouterConfig.RouterDispatcher);
-            var routerMailbox = _system.Mailboxes.CreateMailbox(props, null);
-            //TODO: Should be val routerMailbox = system.mailboxes.getMailboxType(routerProps, routerDispatcher.configurator.config)
+                var lookup = (lookupDeploy ? Deployer.Lookup(path) : null) ?? Deploy.None;
+                var fromProps = new List<Deploy>() { props.Deploy.WithRouterConfig(props.Deploy.RouterConfig.WithFallback(props.Deploy.RouterConfig)), deploy, lookup };
```

Comment:
Are you referring to this crazy-assed thing?

`props.Deploy.WithRouterConfig(props.Deploy.RouterConfig.WithFallback(props.Deploy.RouterConfig))`

Because if so, yeah, we should probably get rid of that - it's something that didn't make a lot of sense in the scala IMPL either.